code/game/machinery/bots/bots.dm

```diff
@@ -82,17 +82,10 @@
 			I.bots += src
 	bots_list += src
 	machines -= src // We have our own subsystem.
-	if (ticker && ticker.current_state == GAME_STATE_PLAYING)
-		initialize()
-
-// Associate the bot with a radio controller created in world/New().
-/obj/machinery/bot/initialize()
-	if(radio_controller)
-		if(bot_flags & BOT_CONTROL)
-			radio_controller.add_object(src, control_freq, filter = control_filter)
-		if(bot_flags & BOT_BEACON)
-			radio_controller.add_object(src, beacon_freq, filter = RADIO_NAVBEACONS)
-	..()
+	if(bot_flags & BOT_CONTROL)
+		radio_controller.add_object(src, control_freq, filter = control_filter)
```

Comment:
Please don't, assuming the ` radio_controller` actually exists in `New()` is very dangerous and it took me plenty of time to figure out it was causing bugs on the live server
Keep this in `initialize()` (assuming you didn't put that block back in `New()` by mistake)

Reply:
I made sure it exists, though. It is now created by the variable declaration.
If I keep this in initialize, it will run after Destroy (because it's being created and destroyed before the game started), causing a hard del due to the radio controller keeping a reference to it, which should have been cleared by Destroy, which ran first.
I suppose an alternative would be to return from initialize() if `src.gcDestroyed`. I can either keep it like it is or do it like I just said.